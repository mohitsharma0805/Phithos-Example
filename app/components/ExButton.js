import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import config from '../config';

const ExButton = ({ buttonStyle, text, onPress }) => {
  return (
    <TouchableOpacity onPress={onPress} style={[styles.button, buttonStyle]}>
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  );
};

export default ExButton;

const styles = StyleSheet.create({
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    marginHorizontal: 20,
    borderRadius: 5,
    height: 50,
    backgroundColor: config.colors.appTheme,
  },
  text: { color: config.colors.white, fontSize: 17 },
});
