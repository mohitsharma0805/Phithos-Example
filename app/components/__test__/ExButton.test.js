import React from 'react';
import { shallow } from 'enzyme';
import { TouchableOpacity, Text } from 'react-native';
import ExButton from '../ExButton';

describe('ExButton', () => {
  describe('Rendering', () => {
    const wrapper = shallow(<ExButton />);
    it('renders the container view', () => {
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find(TouchableOpacity).length).toEqual(1);
      expect(wrapper.find(Text).length).toEqual(1);
    });
  });
  describe('Interaction', () => {
    it('Press event works', () => {
      const onPressEvent = jest.fn();
      const wrapper = shallow(<ExButton onPress={onPressEvent} />);
      wrapper.find(TouchableOpacity).first().props().onPress();
      expect(onPressEvent.mock.calls.length).toBe(1);
    });
  });
});
