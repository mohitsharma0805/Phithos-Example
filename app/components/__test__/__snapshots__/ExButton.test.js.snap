// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExButton Rendering renders the container view 1`] = `ShallowWrapper {}`;
